storage/inmemory/accounts.go

```diff
@@ -0,0 +1,87 @@
+package inmemory
+
+import (
+	"context"
+	"fmt"
+
+	accountspb "github.com/openbank/gunk/gunk/v1/accounts"
+)
+
+type AccountStore struct {
+	data map[string]*accountspb.Account
+	keys []string
+}
+
+// NewAccountStore creates a new inmemory storage
+func NewAccountStore() *AccountStore {
+	return &AccountStore{
+		data: map[string]*accountspb.Account{},
+		keys: []string{},
+	}
+}
+
+// GetAccount returns an account.
+func (s *AccountStore) GetAccount(ctx context.Context, id string) (*accountspb.Account, error) {
+	if a, ok := s.data[id]; ok {
+		return a, nil
+	}
+	return nil, fmt.Errorf("not found")
+}
+
+// GetAccounts returns a list of accounts.
+func (s *AccountStore) GetAccounts(ctx context.Context, nextID string) ([]*accountspb.Account, bool, error) {
+	hasMore := false
+	var sel []string
+	for i, k := range s.keys {
+		if k == nextID {
+			high := i + 10
+			if high > len(s.keys) {
+				high = len(s.keys)
+				hasMore = false
+			} else {
+				hasMore = true
+			}
+			sel = s.keys[i:high]
+		}
+	}
+	var res []*accountspb.Account
+	for _, k := range sel {
+		if a, ok := s.data[k]; ok {
+			res = append(res, a)
+		}
+	}
+	return res, hasMore, nil
+}
+
+// CreateAccount stores an account.
+func (s *AccountStore) CreateAccount(ctx context.Context, account *accountspb.Account) error {
+	if _, ok := s.data[account.AccountID]; ok {
+		return fmt.Errorf("account %s already exists", account.AccountID)
+	}
+	s.keys = append(s.keys, account.AccountID)
+	s.data[account.AccountID] = account
+	return nil
+}
+
+// UpdateAccount updates an account.
+func (s *AccountStore) UpdateAccount(ctx context.Context, account *accountspb.Account) error {
+	if _, ok := s.data[account.AccountID]; ok {
+		s.data[account.AccountID] = account
+		return nil
+	}
+	return fmt.Errorf("not found")
+}
+
+// DeleteAccount removes an account from the storage.
+func (s *AccountStore) DeleteAccount(ctx context.Context, id string) error {
+	if _, ok := s.data[id]; ok {
+		delete(s.data, id)
```

Comment:
Can we use the new Delete method for maps? https://golang.org/pkg/expvar/#Map.Delete

Reply:
@Hekewa does it work with builtin `map`?

Reply:
No it's a method for the expvar type Map.

> Package expvar provides a standardized interface to public variables, such as operation counters in servers. It exposes these variables via HTTP at /debug/vars in JSON format.

I don't think it's what we should use here instead of the builtin map.

Reply:
Instead of the map from `expvar`, I've used `sync.Map` as it provides a concurrent safe map